event_sink_clickhouse/sinks/course_published.py

```diff
@@ -0,0 +1,233 @@
+        modulestore = CoursePublishedSink._get_modulestore()
+        detached_xblock_types = CoursePublishedSink._get_detached_xblock_types()
+
+        dump_id = str(uuid.uuid4())
+        dump_timestamp = str(timezone.now())
+
+        # create a location to node mapping we'll need later for
+        # writing relationships
+        location_to_node = {}
+        items = modulestore.get_items(course_id)
+
+        # create nodes
+        i = 0
+        for item in items:
+            i += 1
+            fields = self.serialize_item(item, i, detached_xblock_types, dump_id, dump_timestamp)
+            location_to_node[self.strip_branch_and_version(item.location)] = fields
+
+        # create relationships
+        relationships = []
+        for item in items:
+            for index, child in enumerate(item.get_children()):
+                parent_node = location_to_node.get(self.strip_branch_and_version(item.location))
+                child_node = location_to_node.get(self.strip_branch_and_version(child.location))
+
+                if parent_node is not None and child_node is not None:
+                    relationship = {
+                        'course_key': str(course_id),
+                        'parent_location': str(parent_node["location"]),
+                        'child_location': str(child_node["location"]),
+                        'order': index,
+                        'dump_id': dump_id,
+                        'time_last_dumped': dump_timestamp,
+                    }
+                    relationships.append(relationship)
+
+        nodes = list(location_to_node.values())
+        return nodes, relationships
+
+    def dump(self, course_key):
+        """
+        Do the serialization and send to ClickHouse
+        """
+        nodes, relationships = self.serialize_course(course_key)
+
+        self.log.info(
+            "Now dumping %s to ClickHouse: %d nodes and %d relationships",
+            course_key,
+            len(nodes),
+            len(relationships),
+        )
+
+        course_string = str(course_key)
+
+        try:
+            # Params that begin with "param_" will be used in the query replacement
+            # all others are ClickHouse settings.
+            params = {
+                # Fail early on bulk inserts
+                "input_format_allow_errors_num": 1,
+                "input_format_allow_errors_ratio": 0.1,
+            }
+
+            # "query" is a special param for the query, it's the best way to get the FORMAT CSV in there.
+            params["query"] = f"INSERT INTO {self.ch_database}.course_blocks FORMAT CSV"
+
+            output = io.StringIO()
+            writer = csv.writer(output, quoting=csv.QUOTE_NONNUMERIC)
+
+            for node in nodes:
+                writer.writerow(node.values())
+
+            response = requests.post(self.ch_url, data=output.getvalue(), params=params, auth=self.ch_auth,
+                                     timeout=self.ch_timeout_secs)
```

Comment:
I think something like this would look better:
```suggestion
            response = requests.post(
                self.ch_url,
                data=output.getvalue(),
                params=params,
                auth=self.ch_auth,
                timeout=self.ch_timeout_secs,
            )
```